[lang] add support for logical and/or expressions

tstack opened the pull request.

rcoh reviewed (commented):
Looks good. Can you add:
- `&&` / `||` as aliases
- tests / negative tests for precedence, eg. `a and b and c and d`
- What happens if there is a field named `and`? I assume it would need to be escaped? Might be nice to show that as a hint if we get `and`/ `or` in a place we don't expect them

tstack commented:
> * `&&` / `||` as aliases

added

> * tests / negative tests for precedence, eg. `a and b and c and d`

I added the above example.  

The original change had the following test for precedence, was there something else you were thinking of?

https://github.com/rcoh/angle-grinder/pull/153/files#diff-9184a440fdb6af807da5a90cdcd5865c62abef73df0a89c2890129ee9e9a011cR2075

And, there's a test for a dangling operator:

https://github.com/rcoh/angle-grinder/pull/153/files#diff-9184a440fdb6af807da5a90cdcd5865c62abef73df0a89c2890129ee9e9a011cR2194

> * What happens if there is a field named `and`? I assume it would need to be escaped? Might be nice to show that as a hint if we get `and`/ `or` in a place we don't expect them

I think one of the benefits of parser combinators is that things like this just kinda work since there's no tokenization step.  I've added this test that calls a function with an argument of `and`:

https://github.com/rcoh/angle-grinder/pull/153/files#diff-9184a440fdb6af807da5a90cdcd5865c62abef73df0a89c2890129ee9e9a011cR2163
